src/platform_impl/windows/window.rs

```diff
@@ -472,27 +472,40 @@ impl Window {
     }
 
     unsafe fn handle_os_dragging(&self, wparam: WPARAM) {
-        let points = {
-            let mut pos = unsafe { mem::zeroed() };
-            unsafe { GetCursorPos(&mut pos) };
-            pos
-        };
-        let points = POINTS {
-            x: points.x as i16,
-            y: points.y as i16,
-        };
-        unsafe { ReleaseCapture() };
+        {
+            let mut guard = self.window_state_lock();
+            if !guard.dragging {
+                guard.dragging = true;
+            } else {
+                return;
+            }
+        }
```

Comment:
Hm, but we're not really dragging here, since we start drag on the main thread. The other code uses `window` and passes `Arc` to state if it matters (see `ime` stuff), I'd suggest to follow it.

To deny drag I think you want to check inside the callback as well, since the callbacks are executed sequentially on the main thread, thus it should correctly suppress when the drag is started.

Reply:
@kchibisov Could you point me a little more concretely? This code was setting this `dragging` boolean before (and I wanted to avoid setting it several times in a row / triggering the callback lots of times and re-starting the drag event).

It's also the only case in which this `dragging` boolean is used from what I can tell, so probably it's just misnamed?

Reply:
Are you suggesting moving this logic into the callback?

Reply:
yes, because `main loop` has a logic to run on the `dragging`, so you should set the variable from the main loop as well the same as the old code did. Just ensure that it's not dragging, I guess.

So, move and check everything inside the callback.

Reply:
I see what you mean; I had missed that logic. Updated the code to reflect this :-).